packages/cli/lib/pipeline.js

```diff
@@ -2,15 +2,71 @@ import { dirname, resolve } from 'path'
 import fromStream from 'rdf-dataset-ext/fromStream.js'
 import rdf from '@zazuko/env'
 import fromFile from 'rdf-utils-fs/fromFile.js'
+import clownface from 'clownface'
 import findPipeline from '../findPipeline.js'
+import discoverManifests from './discoverManifests.js'
+import ns from './namespaces.js'
+
+const discoverOperations = async () => {
+  const ops = rdf.termMap()
+  for await (const { manifest } of discoverManifests()) {
+    manifest
+      .has(rdf.ns.rdf.type, ns.p.Operation)
+      .forEach(x => { // distinguish EcmaScript and EcmaScriptModule?
+        const link = x.out(ns.code.implementedBy).out(ns.code.link).term
+
+        ops.set(x.term, link)
+      })
+  }
+
+  return ops
+}
+
+export const desugarWith = context => dataset => {
+  const { knownOperations } = context
+  const ptr = clownface({ dataset })
+  let n = 0
+  ptr.has(ns.rdf.type, ns.p.Pipeline)
+    .out(ns.p.steps)
+    .out(ns.p.stepList)
+    .forEach(x => {
+      for (const step of x.list()) {
+        if (step.out(ns.code.implementedBy).terms.length === 0) {
```

Comment:
FYI, for that purpose I published the package [is-graph-pointer](https://npm.im/is-graph-pointer)

```
import { isGraphPointer } from 'is-graph-pointer'
```

```suggestion
        if (!isGraphPointer(step.out(ns.code.implementedBy))) {
```